linera-core/src/client.rs

```diff
@@ -1243,9 +1238,11 @@ where
                 })],
             )
             .await?;
-        if !certificate.value.has_effect(&effect_id) {
-            bail!("Failed to open a new chain");
-        }
+        // The second last effect created the new chain.
+        let effect_id = certificate
+            .value
+            .nth_last_effect_id(2)
+            .ok_or_else(|| anyhow!("Failed to open a new chain"))?;
```

Comment:
It's an improvement, but it still feels a bit like relying on an implementation detail. It would feel safer to search, starting from the end, for the `ApplicationCreated` effect.